### DISCUSSION TITLE
Tailwindcss Standalone CLI 3.1.6 does not seem to work for me

### DISCUSSION THREAD

**Discussion by murugappanrm** [tailwindlabs/tailwindcss#8904] [Help]
HI
I tried using the Standalone CLI as suggested. Initially when i define the page and tailwind details and generated the css style sheet using the tailwindcss.exe, it all seems to work great. The problem came when i changed any of the classes, especially color related. The page display incorrect contents in the styling.  The change only gets applied after the generate the style sheet again using tailwindcss.exe. The following are the codes.

**Index.html**

```
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Tailwind-standalone</title>
    <link rel="stylesheet" href="./tailwind.css" />
  </head>
  <body>
    <div class="w-full h-screen bg-primary p-12">
      <div class="prose prose-xl">
        <h1 class="text-clifford">Hi, I am Muru</h1>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Omnis, magni
          incidunt? Facilis temporibus placeat, laborum laboriosam minus
          officia, tenetur deleniti aperiam earum dolorum alias, atque dicta
          facere optio perspiciatis cumque!
        </p>
        <p>
          Lorem ipsum dolor sit amet consectetur, adipisicing elit. Debitis et
          sit perspiciatis, facilis quam ipsa itaque minus sunt nobis, maiores
          libero
          <strong
            >asperiores laboriosam unde autem magnam neque eos minima
            vel?</strong
          >
        </p>
      </div>
      <div class="mt-12">
        <a
          href=""
          class="px-8 py-3 rounded-md bg-[#1DA1F2] shadow-[#1DA1F2]/40 shadow-md text-white hover:bg-sky-700"
          >Follow on twitter</a
        >
      </div>
    </div>
  </body>
</html>
```
**tailwind.css files under the folder tailwindcss**
```

@tailwind base;
@tailwind components;
@tailwind utilities;

@layer components {
  .video {
    @apply px-12 text-center bg-gray-200 shadow-md py-14;
  }
}

@layer utilities {
  .grid-custom {
    grid-template-columns: repeat(3, minmax(0, max-content));
  }
}
@layer components {
  .bg-primary {
    @apply bg-gradient-to-br from-orange-100 to-slate-100 via-red-100;
  }
}
```

**tailwind.config.js**

```
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: ["./*.html"],
  theme: {
    extend: {
      colors: {
        clifford: "#da373d",
      },
    },
  },
  plugins: [
    require("@tailwindcss/typography"),
    // require('@tailwindcss/forms'),
    // require('@tailwindcss/line-clamp'),
    // require('@tailwindcss/aspect-ratio')
  ],
};
```

**Command to generate the stylesheet**

"./tailwindcss -i ./tailwindcss/tailwind.css -o tailwind.css"

**My environment**

Windows 10
Chrome/Firefox

I posted the same under discussions and got no help. So, I am posting it here under issues to get some attention. Appreciate all the help i can get. I have attached a zip file with all the code just in case.

[tailwindcss.zip](https://github.com/tailwindlabs/tailwindcss/files/9149001/tailwindcss.zip)

**Comment by arisac** [comment]
it the correct behavior. you have to regenerate the output css file every-time you build your product.

TailwindCSS is utility CSS, not js. It's using classes, all classes you use will be generated into the output css file.

`npx tailwindcss -i ./src/input.css -o ./dist/output.css --watch` is recommend for dev env to use, `--watch` flag means tailwind will monitor the changes and generate the new css on the fly.

But for you build process of your production environment, you have to generate the css.

**Comment by murugappanrm** [comment]
@arisac 

I am sorry, what you replied does not make sense. I need to generate the styles only if i have made changes to the tailwind.config.js or tailwind.css. Why do i have to generate all the tailwindcss classes each time i just make a simple change e.g bg-yellow-500 to bg-green-500????

If what you say is true, the creators of the Standalone CLI can just dump it into the garbage can.

**Reply by arisac** [reply]
**The Standalone CLI was introduced for people don't have to use node.js or npm in their project.** 

- It generate the css file with the necessary  classes need in the project. And it won't create a `bg-green-500` if it is not in your code. 

- I'm not really sure what you will need TailwindCSS in your project, or dev production procedure is like. But basically the standalone CLI is good for people to generate css file in environment without node.js or npm. And yes, for each build you have to generate the css file.

- The watcher mode: `./tailwindcss -i input.css -o output.css --watch` is good for dev mode when writing your code, it generate the css on the fly, so you don't have to generate the css every-time you made a change during the coding.

However, there is a JS version, TailwindCSS Play CDN to try Tailwind right in the browser without any build step. I don't recommend it because The Play CDN is designed for development purposes only, and is not the best choice for production.

But you can try it https://tailwindcss.com/docs/installation/play-cdn

**Reply by murugappanrm** [reply]
@ arisac,

Now i get the picture. Correct me if i am wrong, Standalone CLI generates css code only for those defined in the scripts like html and js. Its more like Purifycss function built in. Right? 

Ok let me test something out and confirm this.

**Comment by murugappanrm** [comment]
@adamwathan  and @arisac ,

Thank you for the help. With your guidance, i tried some testing and i this is now my understanding and discovery:

(1) Standalone does not function the same as the Tailwindcss CLI.
(2) The first-time generates all the classes required in the HTML
(3) I checked the generated css file and i could not find bg-yellow-500
(4) Then i defined the bg-yellow-500 in the index.html and on generation i found the css for bg-yellow-500.

After @adamwathan question on using '--watch" flag. I decided to the set the flag so that auto generates the css files each time i make a change. 

It great now. Only request is that, as these things are not clear in the documentation, appreciate the documentation is updated at least with a sample tutorial. This will avoid the issue being repeated.

**Reply by adamwathan** [reply]
> (1) Standalone does not function the same as the Tailwindcss CLI.

It behaves exactly the same way, it's the same code. Tailwind generates your CSS on demand, it only generates the CSS you are actually using, so it needs to rebuild any time you change your HTML.

> If what you say is true, the creators of the Standalone CLI can just dump it into the garbage can.

I have zero patience for this so I am blocking you from the repository, just not going to spend any of my time helping people with this attitude.
